src/Kafka/ManageKafkaPermissions/components/ResourceType.stories.tsx

```diff
@@ -0,0 +1,46 @@
+import { Modal, ValidatedOptions, Form } from "@patternfly/react-core";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+
+import { ResourceType } from "./ResourceType";
+
+export default {
+  component: ResourceType,
+  args: {
+    resourceTypeValue: undefined,
+    resourceTypeValidated: ValidatedOptions.default,
+  },
+} as ComponentMeta<typeof ResourceType>;
+
+const Template: ComponentStory<typeof ResourceType> = (args, { id }) => {
+  return (
+    <Modal
+      title="Simple modal header"
+      isOpen={true}
+      appendTo={() => document.getElementById(id) || document.body}
+      disableFocusTrap={true}
+    >
+      <Form>
+        <ResourceType {...args} />
+      </Form>
+    </Modal>
+  );
+};
+
+export const InitialState = Template.bind({});
```

Comment:
```suggestion
export const WorksWithinModals = Template.bind({});
```

This story is more about being sure the dropdown opens ok when in a modal than anything else.
I'd make it more clear with a different name and an updated description.

A story about being able to set an initial value could be named `WithInitialValue`, configured with an initial value other than `undefined`